docs/reference/transform/transforms-at-scale.asciidoc

```diff
@@ -0,0 +1,197 @@
+<<get-transform-stats, {transforms} stats API>>). If, for example, the results 
+show that the highest proportion of  time is spent on search, then prioritize 
+efforts on optimizing the search query of the {transform}. {transforms-cap} also 
+has https://esrally.readthedocs.io[Rally support] that makes it possible to run 
+performance checks on {transforms} configurations if it is required. If you 
+optimized the crucial factors and you still experience performance issues, you 
+may also want to consider improving your hardware.
+
+
+[discrete]
+[[frequency]]
+== 1. Optimize `frequency` (index)
+
+In a {ctransform}, the `frequency` configuration option sets the interval 
+between checks for changes in the source indices when the {transform} runs 
+continuously. If changes are detected, then the source data is searched and the 
+changes are applied to the destination index. Changing the frequency will change 
+the profile of the {transform} workload. Depending on your use case, you may 
+wish to reduce the frequency at which changes are applied. By setting 
+`frequency` to a higher value (maximum is one hour) workload can be spread over 
+time for the cost of less up-to-date data.
+
+
+[discrete]
+[[search-queries]]
+== 2. Profile and optimize your search queries (search)
+
+If you have defined a {transform} source index `query`, ensure it is as 
+efficient as possible. Use the **Search Profiler** under **Dev Tools** in {kib} 
+to get detailed timing information about the execution of individual components 
+in the search request. Alternatively, you can use the <<search-profile>>. The 
+results give you insight into how search requests are executed at a low level so 
+that you can understand why certain requests are slow, and take steps to improve 
+them.
+
+{transforms-cap} execute standard {es} search requests. There are different ways 
+to write {es} queries, and some of them are more efficient than others. Consult 
+<<tune-for-search-speed>> to learn more about {es} performance tuning.
+
+
+[discrete]
+[[index-sorting-group-by-ordering]]
+== 3. Use Index Sorting and `group_by` ordering (search, process)
+
+If you use more than one `group_by` field in your {transform}, then the order of 
+the fields in conjunction with the use of <<index-modules-index-sorting>> may 
+improve runtime.
+
+Index sorting enables you to store documents on disk in a specific order which 
+can improve query efficiency. The ideal sorting logic depends on your use case, 
+but the rule of thumb may be to sort the fields in descending order (high to low 
+cardinality) starting with the time-based fields. Then put the time-based 
+components first in the `group_by` if you have any, and then apply the same 
+order to your `group_by` fields as configured for index sorting. 
+
+
+[discrete]
+[[increase-shards-dest-index]]
+== 4. Increase the number of shards of the destination index (index)
+
+<<scalability,Depending on the size of the destination index>>, you may consider 
```

Comment:
> <<scalability,Depending on the size of the destination index>>, you may consider 

I think this link should be referenced separately, since I'm not sure why I'd click it here. Maybe something like this:

> Depending on the size of the index, you may consider increasing its shard count... create the destination index before you start the transform. For more information about how the number of shards affects scalability and resilience, refer to <<scalability>>.